compatibility/compatibility-vec12to20/src/main/java/com/foursoft/harness/compatibility/vec12to20/util/Vec20XReader.java

```diff
@@ -0,0 +1,97 @@
+ * %%
+ * Copyright (C) 2020 - 2023 4Soft GmbH
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.harness.compatibility.vec12to20.util;
+
+import com.foursoft.harness.compatibility.vec.common.VecCreator;
+import com.foursoft.harness.compatibility.vec.common.VecProcessor;
+import com.foursoft.harness.compatibility.vec12to20.Vec12XTo20XProcessor;
+
+import java.io.InputStream;
+
+/**
+ * Class which is able to read and create any VEC 1.X.X.
+ */
+public final class Vec20XReader {
+
+    private static final VecProcessor CREATOR = new Vec12XTo20XProcessor();
+
+    private Vec20XReader() {
+        //hide default constructor
+    }
+
+    /**
+     * This method creates a VecContent in version 2.0.X from the given {@link InputStream}.
+     *
+     * @param inputStream InputStream of a VEC.
+     * @param streamName  The name of the stream for logging.
+     * @return The VecContent in version 2.0.X.
+     */
+    public static com.foursoft.harness.vec.v2x.
+            VecContent createVec20x(final InputStream inputStream, final String streamName) {
+        return new VecCreator(CREATOR).createVec(inputStream, streamName,
+                                                 com.foursoft.harness.vec.v2x.VecContent.class);
+    }
+
+    /**
+     * This method creates a VecContent in version 1.2.X from the given {@link InputStream}.
+     *
+     * @param inputStream InputStream of a VEC.
+     * @param streamName  The name of the stream for logging.
+     * @return The VecContent in version 1.2.X.
+     */
+    public static com.foursoft.harness.vec.v12x.
+            VecContent createVec12x(final InputStream inputStream, final String streamName) {
+        return new VecCreator(CREATOR).createVec(inputStream, streamName,
+                                                 com.foursoft.harness.vec.v12x.VecContent.class);
+    }
```

Comment:
In my opinion, this should come first / the methods should be in the same order (vec12, vec2, vec12, vec2).